src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1766,6 +1786,73 @@ class SetClauseSegment(BaseSegment):
     )
 
 
+@ansi_dialect.segment()
+class FunctionDefinitionGrammar(BaseSegment):
+    """This is the body of a `CREATE FUNCTION AS` statement."""
+
+    match_grammar = Sequence(
+        "AS",
+        Ref("QuotedLiteralSegment"),
+        Sequence(
+            "LANGUAGE",
+            # Not really a parameter, but best fit for now.
+            Ref("ParameterNameSegment"),
+            optional=True,
+        ),
+    )
+
+
+@ansi_dialect.segment()
+class CreateFunctionStatementSegment(BaseSegment):
+    """A `CREATE FUNCTION` statement.
+
+    This version in the ANSI dialect should be a "common subset" of the
```

Comment:
I think we should have it somewhere in the docs, maybe for contributors or inside Architecture, specified that the ANSI dialect is our base dialect and it hosts logic that is common across multiple languages, so it does not strictly represent the ANSI specification

Reply:
Will make a new issue for this.